Tests/ComposableArchitectureTests/DebugTests.swift

```diff
@@ -373,6 +373,22 @@ final class DebugTests: XCTestCase {
     )
   }
 
+  func testRecursiveOutput() {
+    class Foo {
+      var foo: Foo?
+    }
+    let foo = Foo()
+    foo.foo = foo
+    XCTAssertEqual(
+      debugOutput(foo),
+      """
+      Foo(
+        foo: Foo()
```

Comment:
Not sure if we want to show somehow _which_ object this was...like:

```
Foo#1(
  foo: Foo#1()
)
```
Or if you have other ideas.

Reply:
huh, I'm down to leave as-is until we find a reason to change.

Reply:
Some other thoughts (won't block merging, but could enhance later):

```
Foo(
  foo: Foo(...)
)
```

```
Foo(
  foo: Foo(↩︎)
)
```